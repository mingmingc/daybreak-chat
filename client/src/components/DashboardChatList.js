import React from 'react';
import { USER_ID } from '../const';
import { Link } from 'react-router-dom';

const DashboardChatList = (props) => {
    const channels = props.channels;

    const friendName = (channel) => {
        if (channel.members[0].userId !== USER_ID) {
            return channel.members[0].userId;
        } 
        return channel.members[1].userId;
    }

    let chatList = channels.map((channel => {
        return (
            <div className="chat card" key={friendName(channel)}>
                <div className="card-content">
                    <p className="title">{friendName(channel)}</p>
                    <Link to="/chat">Go to chat</Link>
                </div>
            </div>
        );
    }))

    return (chatList);
}

export default DashboardChatList;