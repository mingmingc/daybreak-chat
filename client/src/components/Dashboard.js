import React, { Component } from 'react';
import { USER_ID } from '../const';
import { _getSbInstance } from '../utils';
import DashboardChatList from './DashboardChatList';

class Dashboard extends Component {
    constructor(props) {
        super(props);
        this.state = {
            channels: []
        }
        this.userId = USER_ID;
        this.sb = _getSbInstance();

        this.getChannels = this.getChannels.bind(this);
    }

    componentDidMount() { 
        this.getChannels();
    }

    getChannels() {
        let allUserChannels = this.sb.GroupChannel.createMyGroupChannelListQuery();
        allUserChannels.includeEmpty = true;
        allUserChannels.userIdsIncludeFilter = [this.userId];
        return new Promise((resolve, reject) => {
            allUserChannels.next((myChannels, error) => {
              if (error) {
                reject(error);
              } else {
                resolve(myChannels);
                console.log(myChannels);
                this.setState({channels: myChannels})
              }
            });
        });
    }
     
    render() {
        return(
            <div className="Dashboard">
                <h1 className="title is-1"> Welcome, {this.userId} </h1>
                <DashboardChatList 
                    channels = {this.state.channels}
                    openChatHandler = {this.openChatHandler}
                /> 
            </div>
        )
    }
}

export default Dashboard;