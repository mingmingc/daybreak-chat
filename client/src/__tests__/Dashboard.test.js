import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import '../setupTests';
import Dashboard from '../components/Dashboard';
import DashboardChatList from '../components/DashboardChatList';

jest.mock('../components/Dashboard'); 
jest.mock('../components/DashboardChatList'); 

it('invokes `componentDidMount` when mounted', () => {
    jest.spyOn(Dashboard.prototype, 'componentDidMount');
    shallow(<Dashboard />);
    expect(Dashboard.prototype.componentDidMount).toHaveBeenCalled();
    Dashboard.prototype.componentDidMount.mockRestore();
});

const testChannels = [
    { members: [
            { name: "testFriend", url: "" },
            { name: "testFriend1", url: "" }
        ]
    },
    { members: [
            { name: "testFriend2", url: "" },
            { name: "testFriend3", url: "" }
        ]
    }
]

describe('getChannels', () => {
    const wrapper = shallow(<Dashboard />);
    const instance = wrapper.instance();
    jest.spyOn(instance, 'getChannels');
    instance.componentDidMount();
    it('calls `getChannels` when mounted', () => {
        expect(instance.getChannels).toHaveBeenCalled();
    });
    it('returns promise', () => {
        expect(instance.getChannels).toHaveReturned();
    });
});

;

// describe('Render all channels', () => {
//     test('it should render expected number of channels', () => {
//         shallow(<Dashboard channels={testChannels}><DashboardChatList /></Dashboard>);
//         expect(document.querySelectorAll('.chat').length).toBe(testChannels.length)
//     });
// });