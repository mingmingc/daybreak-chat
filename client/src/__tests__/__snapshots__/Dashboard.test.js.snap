// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should call function 1`] = `
<div
  className="Dashboard"
>
  <h1
    className="title is-1"
  >
     Welcome, 
    testUser
     
  </h1>
</div>
`;
